### PR TITLE
Changelog 4.46.0

### LINKED ISSUE #3433: [Documentation] Change changelog.txt to .md and update NuGet package to point to releases page
**Description of the issue**
Change changelog.txt to .md and update NuGet package to point to releases page (same as in Microsoft Identity Web).

### PR DESCRIPTION
**Pull request by pmaytak**
Fixes #3433

**Changes proposed in this request**
- Move from changelog.txt to changelog.md.
- Add 4.46 section.
- Fix typos in 4.45 section.
- Update changelog information in project properties.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -0,0 +1,1164 @@
+4.46.0
+==========
+
+### New Features
+**Added `AcquireTokenByUsernamePassword` flow in WAM broker preview**. See [3308](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3308).  
+**Added support for Proof-of-Possession tokens to `AcquireTokenByUsernamePassword` flow in WAM broker preview**. See [3308](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3308).  
+**Added `WithTenantIdFromAuthority` API to request builder**. See [3429](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3429).  
+**Exposed new Identity Logger in the `TokenCacheNotificationArgs`**. See [3404](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/3404).  
+**Increased security of PKCE**. See [1777](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1777).
```
Comment (bgavrilMS) [inline changelog.md:9]:
```suggestion
**[Security] Increased size of PKCE verifier**. See [1777](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1777).  
```

**Review by bgavrilMS — APPROVED** [review]

**Comment by MichelZ** [comment]
I don't think the page looks particularly good...
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/8fc5b7f286821b3ac5fbd1680b1847e5c6594e59/changelog.md

- Why does everything need to be bold?
- Shouldn't it use a list? ("-")

**Review by MichelZ — COMMENTED** [review]
Nit: use uppercase CHANGELOG.md

**Review by neha-bhargava — APPROVED** [review]

**Comment by pmaytak** [comment]
> I don't think the page looks particularly good... https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/8fc5b7f286821b3ac5fbd1680b1847e5c6594e59/changelog.md
> 
> * Why does everything need to be bold?
> * Shouldn't it use a list? ("-")

That's just the style we've been using for a while now. I think originally the bold part of each item was like a title and then there was non-bold description. But then we just started putting all the details into the linked issues themselves. I'm fine with using a bullet list wit non-bold text, if it makes it more readable, unless others have objections.

**Review by gladjohn — APPROVED** [review]
looks good

**Comment by MichelZ** [comment]
> > I don't think the page looks particularly good... https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/8fc5b7f286821b3ac5fbd1680b1847e5c6594e59/changelog.md
> > 
> > * Why does everything need to be bold?
> > * Shouldn't it use a list? ("-")
> 
> That's just the style we've been using for a while now. I think originally the bold part of each item was like a title and then there was non-bold description. But then we just started putting all the details into the linked issues themselves. I'm fine with using a bullet list wit non-bold text, if it makes it more readable, unless others have objections.

Nice, thanks. That does make it much more readable!

**Pull request merged**